ocaml/sdk-gen/java/autogen/xen-api/src/main/java/com/xensource/xenapi/Connection.java

```diff
@@ -29,228 +29,211 @@
+     * passing the Connection as a parameter. No attempt to connect to the server is made until login is called.
+     * <p>
+     * When this constructor is used, a call to dispose() will do nothing. The programmer is responsible for manually
+     * logging out the Session.
      *
-     * @param url The URL of the server to connect to
-     * @param sessionReference A reference to a logged-in Session. Any method calls on this Connection will use it.
-     *                         This constructor does not call Session.loginWithPassword, and dispose() on the resulting
-     *                         Connection object does not call Session.logout. The programmer is responsible for
-     *                         ensuring the Session is logged in and out correctly.
-     * @param replyTimeout The reply timeout for xml-rpc calls in seconds
-     * @param connTimeout The connection timeout for xml-rpc calls in seconds
+     * @param url               The URL of the server to connect to. Should be of the form http(s)://host-url./jsonrpc or http(s)://host-url.
+     * @param sessionReference  A reference to a logged-in Session. Any method calls on this Connection will use it.
+     *                          This constructor does not call Session.loginWithPassword, and dispose() on the resulting
+     *                          Connection object does not call Session.logout. The programmer is responsible for
+     *                          ensuring the Session is logged in and out correctly.
+     * @param requestTimeout    The reply timeout for JSON-RPC calls in seconds
+     * @param connectionTimeout The connection timeout for JSON-RPC calls in seconds
      */
-    public Connection(URL url, String sessionReference, int replyTimeout, int connTimeout)
-    {
-        this.client = getClientFromURL(url, replyTimeout, connTimeout);
+    public Connection(URL url, String sessionReference, int requestTimeout, int connectionTimeout) {
+        this.client = new JsonRpcClient(url, requestTimeout, connectionTimeout);
         this.sessionReference = sessionReference;
     }
 
-    private XmlRpcClientConfigImpl config = new XmlRpcClientConfigImpl();
-
-    public XmlRpcClientConfigImpl getConfig()
-    {
-        return config;
+    /**
+     * Updated when Session.login_with_password() is called.
+     */
+    public APIVersion getAPIVersion() {
+        return apiVersion;
     }
 
-    private XmlRpcClient getClientFromURL(URL url, int replyWait, int connWait)
-    {
-        config.setTimeZone(TimeZone.getTimeZone("UTC"));
-        config.setServerURL(url);
-        config.setReplyTimeout(replyWait * 1000);
-        config.setConnectionTimeout(connWait * 1000);
-        XmlRpcClient client = new XmlRpcClient();
-        client.setConfig(config);
-        return client;
+    private void setAPIVersion(Session session) throws IOException {
+        try {
+            long major = session.getThisHost(this).getAPIVersionMajor(this);
```

Comment:
Not caused by this PR, but we have 4 requests here to get the API version which increases the "connection" time. If we do it like the C# SDK, i.e. get the pool records, then the coordinator record, we achieve the same with 2 requests.